frontend/src/pages/NewStudent.vue

```diff
@@ -141,19 +146,41 @@ export default {
               .indexOf(this.nativeLanguage.toLowerCase()) >= 0
             })
     },
-      filteredSourceDataArray() {
+    filteredSourceDataArray() {
         return this.sourceData.filter((sourceOption) => {
             return sourceOption
               .toString()
               .toLowerCase()
               .indexOf(this.source.toLowerCase()) >= 0
             })
-    }  
+    },
+    
   },
 
-
   methods: {
     createStudent(){
+       const studentCreate = {
+        method: "POST",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          PhoneNumber: this.PhoneNumber,
+
+          // Replace FirstName and LastName with Name when Student Model is changed
+          FirstName: this.name,
+          LastName: 'Hossein', // Placeholder as I haven't split Name 
+          Source: this.source,
+          NativeLanguageID: 1, // how to avoid hardcoding the languageID? can the backend process text then match to the appropriate ID? 
+          EnglishProficiency: this.EnglishProficiency, // must be No, Little, Simple or Intermediate. form input is int 
+          Notes: this.Notes,
+          StatusID: 1, //How to avoid hardcoding StatusID? Let the backend handle it?
+        })
+      }
+      fetch("/api/students", studentCreate)
+        .then(response => response.json()) 
+      //Pop-up notification that new student has been added
       this.$buefy.notification.open({
```

Comment:
To show the notification after the successful backend response:
```js
fetch("api/students", studentCreate)
  .then(() => {
    this$buefy.notification.open( ... )
  })
```

Reply:
This comment still stands - we should only show the notification once we know the creation was successful.

Reply:
thanks thomas!! I've refactored my code to try to do that, but it somehow shows a notification + redirects even if the whole form is blank. 

I've reviewed the PR for NativeLanguageString (#42) and will refactor my code in light of that. It works great, just that it doesn't work for PATCH (but it doesn't affect this PR).

Reply:
@szktm this depends on where we do the validation. If you don't want to 'allow' an empty form, then we can check that before calling the fetch method. Even if we do call the fetch with empty data, does the backend return a non-200 response?